### ISSUE TITLE
Register specific diagnostic run actions with "Fix All" action?

### ISSUE THREAD

**Issue by mgzenitech** [tamayika/vscode-any-lint#27]
Would it be possible to register specific diagnostic run actions with "Fix All" action? 

![Screenshot from 2024-07-22 09-20-47](https://github.com/user-attachments/assets/5946d67b-1aac-42e7-875c-5ec99435da54)

Most of the linter extensions integrate well with that.

For example:
```json
          {
            "args": [
              "`fix`",
              "`-q`",
              "`--no-backup`",
              "$.file"
            ],
            "binPath": "`dotenv-linter`",
            "cwd": "$.workspaceFolder",
            "lintAfterRun": "true",
            "title": "`Fix all fixable problems`",
            "type": "run"
          }
```
I'd like this to get called when calling Fix All action. In addition then there's a way to setup VSCode to automatically fix all linting issues on save.

Or exposing some diagnostic run actions would also work, for example how ESLint does that in addition to normal integration:
![Screenshot from 2024-07-22 09-25-52](https://github.com/user-attachments/assets/7ec02240-78d5-4345-ac6a-dbc82926add9)

**Comment by tamayika** [comment]
Maybe you should use this extension for custom command.
https://marketplace.visualstudio.com/items?itemName=edonet.vscode-command-runner

**Comment by mgzenitech** [comment]
> Maybe you should use this extension for custom command. https://marketplace.visualstudio.com/items?itemName=edonet.vscode-command-runner

What's the id for linter diagnostic run action? because I have another extension which allows calling internal VSCode actions.

For example that's how I call other extension action:
```json
  "runOnSave.commands": [
    {
      "async": false,
      "commandBeforeSaving": "stylelint.executeAutofix",
      "match": ".*\\.css$",
      "runIn": "vscode"
    }
  ],
```

`stylelint.executeAutofix` is command id.

In VSCode actions list I see run action but not sure if that's the one to call or even how to use it to trigger my command
![Screenshot from 2024-07-22 15-04-40](https://github.com/user-attachments/assets/4c0a4774-6110-473a-99d5-7ee1f16528a3)

**Comment by tamayika** [comment]
It's not intended to run internal command from another extension.

How about adding one more linter before actual linter if you want to implement auto fix on save?

```jsonc
{
  "any-lint.linters": [
    {
      "name": "dot-env auto fix",
      "condition": "$.fileExtname == '.env'",
      "args": [
        "fix",
        "-q",
        "--no-backup",
        "${file}"
      ],
      "binPath": "dotenv-linter",
      "cwd": "${workspaceFolder}",
      "on": [
        "save"
      ]
    },
    {
      // same as dot-env example
    }
  ]
}
```

**Comment by mgzenitech** [comment]
> It's not intended to run internal command from another extension.
> 
> How about adding one more linter before actual linter if you want to implement auto fix on save?
> 
> ```js
> {
>   "any-lint.linters": [
>     {
>       "name": "dot-env auto fix",
>       "condition": "$.fileExtname == '.env'",
>       "args": [
>         "fix",
>         "-q",
>         "--no-backup",
>         "${file}"
>       ],
>       "binPath": "dotenv-linter",
>       "cwd": "${workspaceFolder}",
>       "on": [
>         "save"
>       ]
>     },
>     {
>       // same as dot-env example
>     }
>   ]
> }
> ```

Yeah, I could just call the command directly but wouldn't it be nicer and somehow more native to be able to integrate actions with `source.fixAll.any-lint` ? Interface could be just by adding additional property, like `"asFixAll": true` ? 

```json
          {
            "args": [
              "`fix`",
              "`-q`",
              "`--no-backup`",
              "$.file"
            ],
            "binPath": "`dotenv-linter`",
            "cwd": "$.workspaceFolder",
            "lintAfterRun": "true",
            "title": "`Fix all fixable problems`",
            "type": "run",
            "asFixAll": true,
          }
```

Or at least have ability to call them directly, set shortcuts, etc
```
any-lint.dotenv-linter.action.(some id that people can add to actions)
```
